### PR TITLE
[feat] Check token status

### PR DESCRIPTION
**Pull request by jiyoung-an**
## Summary
- rename `chat` class to `ai` class.
- add `getUserTokensByStatus` method.  
  - status: `minted`, `assistant_created`, `thread_created`, `message_created`
  - example:
```ts
// if you want to get tokens with status
const result = await ainft.ai.getUserTokensByStatus(objectId, address);

// if you want to get tokens filtered by status
const result = await ainft.ai.getUserTokensByStatus(objectId, address, 'minted');
``` 
```
{
          "total": 2,
          "items": [
              {
                  "tokenId": "7",
                  "owner": "0x7339Ac147C1EE853956cC9E30e97891280D1F8e5",
                  "tokenURI": "https://metadata-api-dev.ainetwork.ai/metadata/ainft721_0x9facba20ed898fc12e6f65063a18d67499ef60ef/MYOBJ/7",
                  "status": "minted"
              },
              {
                  "tokenId": "1",
                  "owner": "0x7339Ac147C1EE853956cC9E30e97891280D1F8e5",
                  "tokenURI": "https://metadata-api-dev.ainetwork.ai/metadata/ainft721_0x9facba20ed898fc12e6f65063a18d67499ef60ef/MYOBJ/1",
                  "status": "minted"
              }
          ]
      }
```
- minor refactoring..

### CODE REVIEW

File: src/ai/ai.ts [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+  }
+
+  /**
+   * @todo Enable deposit function once withdrawal is fully implemented and tested.
+   * Deposits a credits for a service.
+   * @param {string} serviceName - The name of Ainize service.
+   * @param {number} amount - The amount of credits to deposit.
+   * @returns {Promise<CreditTransactionResult>} Returns a promise that resolves with the deposit transaction info.
+   */
+  /*
```
Comment (platfowner) [inline src/ai/ai.ts:84]: could you add a todo describing this commenting or when it will be uncommented?

File: src/ai/ai.ts [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+    const tokens = Object.entries(allTokens).reduce<NftToken[]>((acc, [id, token]) => {
+      if (token.owner === address) {
+        acc.push({ tokenId: id, ...token });
+      }
+      return acc;
+    }, []);
+
+    tokens.map((token) => {
+      let status = 'minted';
```
Comment (platfowner) [inline src/ai/ai.ts:124]: how about adding an enum definition for this?

File: src/ai/ai.ts (hunk 1 above)
Comment (jiyoung-an) [reply src/ai/ai.ts:84]: I added @todo field in the comments. ([line:78])(https://github.com/ainft-team/ainft-js/pull/286/files/7de43d78aa0cbb1e1016b78feacf2168f647710c#diff-ac1856ed663bd0d5ea0185f6587d50114a1ef051bccc90f663627ca8fb9ed9d8R78)

File: src/ai/ai.ts (hunk 2 above)
Comment (jiyoung-an) [reply src/ai/ai.ts]: done!

File: src/ai/ai.ts [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+import { Path } from '../utils/path';
+import { validateObject, validateObjectOwner } from '../utils/validator';
+import { DEFAULT_AINIZE_SERVICE_NAME } from '../constants';
+
+/**
+ * This class manages ai configurations for AINFT object,\
+ * manages the credits needed for their uses, and checks if tokens are available for creation.\
+ * Do not create it directly; Get it from AinftJs instance.
+ */
+export class Ai extends FactoryBase {
```
Comment (platfowner) [inline src/ai/ai.ts:25]: minor) if this class is used for chatbot usecases only (e.g. 'ImageGen' class for image generation AI), you can consider 'Persona' as a name, but it's basically up to you. :)

**Review by platfowner — APPROVED** [review]
LGTM with some comments.
Thanks!

File: src/ai/ai.ts (hunk 3 above)
Comment (jiyoung-an) [reply src/ai/ai.ts:25]: it's a good idea, but i think this class is used in general ai use cases (e.g. configuration, credit, utils)

**Comment by jiyoung-an** [comment]
Thank you for review! now merging..

**Pull request merged**